scripts/zones/Aht_Urhgan_Whitegate/npcs/Lathuya.lua

```diff
@@ -50,5 +143,30 @@ function onEventFinish(player, csid, option)
             var = { OmensProgress }
         })
         player:delKeyItem(tpz.ki.SEALED_IMMORTAL_ENVELOPE)
+
+    -- TRANSFORMATIONS
+    elseif csid == 732 + csOffset then
+        player:setCharVar("CraftingBluArmor", 1)
+        player:confirmTrade()
+    elseif csid == 734 + csOffset then
+        player:setCharVar("CraftingBluArmor", 2)
+        npcUtil.giveKeyItem(player, tpz.ki.MAGUS_ORDER_SLIP)
+        player:setCharVar("BluPaymentDay", vanaDay())
+        player:confirmTrade()
+    elseif csid == 736 + csOffset then
+        if npcUtil.giveItem(player, craftingItems[currentTask][7]) then
+            if currentTask == 1 then
+                player:setMaskBit(player:getCharVar("LathuyaCraftingList_Mask"), "LathuyaCraftingList_Mask", 0, true)
+            elseif currentTask == 2 then
+                player:setMaskBit(player:getCharVar("LathuyaCraftingList_Mask"), "LathuyaCraftingList_Mask", 1, true)
+            elseif currentTask == 4 then
+                player:setMaskBit(player:getCharVar("LathuyaCraftingList_Mask"), "LathuyaCraftingList_Mask", 2, true)
+            end
```

Comment:
Then `[BLUAF]Available` could be set to `[BLUAF]Available - (2 ^ (currentTask - 1))`

(Or: You wouldn't need a "positive-oriented" list of things you've crafted, because you instead have a "negative-oriented" list of things remaining to _be_ crafted)

This would mean adjusting the final piece to have a key of 3 in your craftingItems table (so all the pieces / tasks line up). I want to say this shouldn't be a problem, as `LathuyaCraftingList_Mask` seems to only be used as a table key (and mask bit manipulation which we're changing), and isn't used for any CS or param offsets (you already calculate those based on the total number of pieces crafted).